### ISSUE TITLE
[Inception-Q] Dealing with imbalanced input

### ISSUE THREAD

**Issue by acerekovic** [tensorflow/models#231]
Hi,

This is a question, not an issue, I did not know where to ask. 

When doing transfer learning for flower data, [data/download_and_preprocess_flowers.sh](https://github.com/tensorflow/models/blob/master/inception/inception/data/download_and_preprocess_flowers.sh) first selects random 100 images from the train directory and moves them to validation directory, and then it creates `TFRecord` data. This results with imbalanced train dataset. 

My question would be, how does the training of the inception-v3 model deal with a dataset that is extremely imbalanced, and has very low number of samples in one of the classes ? For example, let's say we have 4-class problem with 110, 1500, 8000 and 2000 examples. Do users have to implement weighted loss function, such as explained [here](http://stackoverflow.com/questions/35155655/loss-function-for-class-imbalanced-binary-classifier-in-tensor-flow), or does inception training have some internal mechanism that already deals with imbalanced data, as example [data/download_and_preprocess_flowers.sh](https://github.com/tensorflow/models/blob/master/inception/inception/data/download_and_preprocess_flowers.sh) (weakly) suggest?

**Comment by prb12** [comment]
This is a question better suited for StackOverflow. Please ask it there and tag it with the `tensorflow` tag.
@shlens Is there a more suitable discussion forum for this sort of thing?

**Comment by shlens** [comment]
The StackOverflow [page](http://stackoverflow.com/questions/35155655/loss-function-for-class-imbalanced-binary-classifier-in-tensor-flow) has a nice description for to handle class label imbalance. I would suggest a quick search on Google for "class label imbalance" (one nice [result](http://machinelearningmastery.com/tactics-to-combat-imbalanced-classes-in-your-machine-learning-dataset/)) as well as on [Google Scholar](https://scholar.google.com/scholar?hl=en&q=class+label+imbalance&btnG=&as_sdt=1%2C5&as_sdtp=).

**Comment by acerekovic** [comment]
OK, thanks a lot, then I'll do some experiments with regards [inception training] (https://github.com/tensorflow/models/blob/master/inception/inception/inception_train.py) and my dataset. Also, thanks for the links, indeed there are some nice results of a simple Google search.
